### PR TITLE
SDIT-1877 Process any CSIP deleted or CSIP Factor Delete - ignore auditModule

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPPrisonOffenderEventListener.kt [hunk 1]
```diff
@@ -71,6 +72,10 @@ class CSIPPrisonOffenderEventListener(
         RETRY_SYNCHRONISATION_MAPPING.name -> csipSynchronisationService.retryCreateCSIPReportMapping(
           sqsMessage.Message.fromJson(),
         )
+
+        RETRY_CSIP_FACTOR_SYNCHRONISATION_MAPPING -> csipFactorSynchronisationService.retryCreateCSIPFactorMapping(
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPPrisonOffenderEventListener.kt:76]: Ensure pick up retries

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationService.kt [hunk 2]
```diff
@@ -89,11 +90,6 @@ class CSIPFactorSynchronisationService(
         "nomisCSIPFactorId" to event.csipFactorId,
       )
 
-    if (event.auditModuleName == "DPS_SYNCHRONISATION") {
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPFactorSynchronisationService.kt:92]: Previous PR comments highlighted we won't know where the delete was coming from - so this check is irrelevant

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPPrisonOffenderEventListener.kt (hunk 1 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/csip/CSIPPrisonOffenderEventListener.kt:76]: Are you missing a test here than? Where mapping create fails once? I think that is the only "unhappyish" path worth doing

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**